### ISSUE TITLE
meta-boxes-product.js throwing error for product attributes

### ISSUE THREAD

**Issue by vendidero** [woocommerce/woocommerce#22673]
As a plugin developer I'm using the action `woocommerce_after_product_attribute_settings` to add another checkbox to the product attributes options. Unfortunately the JS after the AJAX refresh (https://github.com/woocommerce/woocommerce/blob/master/assets/js/admin/meta-boxes-product.js#L473) depends on a static number (6) of input fields. In case further checkboxes or input fields are added the index calculation is broken. 

**To Reproduce**
1. Add a checkbox via `woocommerce_after_product_attribute_settings`
2. Save product attributes 
3. See error in console

I guess we need to dynamically calculate the number of input fields within the (first) product attributes element, e.g.: 

```javascript
var original_data_nr = $( '.product_attributes > .woocommerce_attribute:first' ).find( 'input, select, textarea' ).length;
```

Cheers

**Comment by vendidero** [comment]
Hi there,

this is the HTML I'm currently using:

```
<tr>
            <td>
                <label><input type="checkbox" class="checkbox" <?php checked( $gzd_product_attribute->is_checkout_visible(), true ); ?> name="attribute_checkout_visibility[<?php echo esc_attr( $i ); ?>]" value="1" /> <?php esc_html_e( 'Visible on checkout', 'woocommerce-germanized' ); ?></label>
            </td>
</tr>
```

The `original_data` variable contains all inputs, selects and textareas. This line divides the length by 6 (statically) and will force errors:

`var nr_elements = original_data.length / 6;`

Cheers

**Comment by vendidero** [comment]
Hi @rrennick,

did you check that again? I could provide a pull request for that issue too..

Cheers

**Comment by rrennick** [comment]
@vendidero I tested with both

`<label><input type="checkbox" class="checkbox" checked="checked" name="attribute_checkout_visibility[1]" value="1" /> Visible on checkout</label>
`
and
`<label><input type="checkbox" class="checkbox" name="attribute_checkout_visibility[1]" value="1" /> Visible on checkout</label>
`

No browser console errors & the attribute name/value updated correctly.

**Comment by vendidero** [comment]
@rrennick I've tested that again. See the screenshot for further info:

![bildschirmfoto 2019-02-13 um 15 37 03](https://user-images.githubusercontent.com/8328938/52719156-5abdac80-2fa5-11e9-891f-60f2efa9dc5d.png)

During my testes `original_data` contains 7 inputs (length = 7). This is divided through 6 which results in `1.1666666666666667`. The index calculation within the loop (e.g. `i * 6 + 2`) results in 8 for the second loop which will then throw the error as `original_data` does not contain an element at position 8.

I've tested that by:

1. Edited a product
2. Added a product attribute
3. Save attributes

Cheers
